src/section/section_proof_chain.rs

```diff
@@ -85,7 +89,7 @@ impl SectionProofChain {
     ///
     /// Note: unlike `std::slice`, if the range is invalid or out of bounds, it is silently adjusted
     /// to the nearest valid range and so this function never panics.
-    pub(crate) fn slice<B: RangeBounds<u64>>(&self, range: B) -> Self {
+    pub fn slice<B: RangeBounds<u64>>(&self, range: B) -> Self {
```

Comment:
Worth expanding this comment a bit, now the function is `pub`. Maybe
"Returns a subset of this chain specified by the given index range" or something like that.